### PR TITLE
[Regression Tests][rocRAND] Implementation of regression subtest for emulation

### PR DESCRIPTION
**Pull request by NguyenNhuDi**
The implementation of regression test for emulation in rocRAND.

This is a continuation of (#573), do not merge until (#573 ) is merged and closed.

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -3,6 +3,19 @@
 Documentation for rocRAND is available at
 [https://rocm.docs.amd.com/projects/rocRAND/en/latest/](https://rocm.docs.amd.com/projects/rocRAND/en/latest/)
 
+
+## (Unreleased) rocRAND 3.2.0 for ROCm 6.4
+
+### Added
+* Added regression tests to `rtest.py`. Regression tests are a subset of tests that caused hardware problems for past emulation environments.
+  * Can be ran with `python rtest.py [--emulation|-e|--test|-t]=regression`
```
Comment (amd-jnovotny) [inline CHANGELOG.md]:
```suggestion
  * Can be run with `python rtest.py [--emulation|-e|--test|-t]=regression`
```

File: CHANGELOG.md [hunk 2]
```diff
@@ -3,6 +3,19 @@
 Documentation for rocRAND is available at
 [https://rocm.docs.amd.com/projects/rocRAND/en/latest/](https://rocm.docs.amd.com/projects/rocRAND/en/latest/)
 
+
+## (Unreleased) rocRAND 3.2.0 for ROCm 6.4
+
+### Added
+* Added regression tests to `rtest.py`. Regression tests are a subset of tests that caused hardware problems for past emulation environments.
```
Comment (amd-jnovotny) [inline CHANGELOG.md:10]: This sentence is a bit confusing. Did the regression tests cause the hardware problems? Or do they validate scenarios that have caused problems in the past? Maybe if you describe the situation, we can reword this so it's clearer.

File: CHANGELOG.md [hunk 3]
```diff
@@ -3,6 +3,19 @@
...
+  * Can be ran with `python rtest.py [--emulation|-e|--test|-t]=regression`
+* Added smoke test options, which runs a subset of the unit tests and ensuring that less than 2gb of VRAM will be used
+  * Smoke tests can be ran with `[--emulation|-e|--test|-t]=smoke`
+* Added `--emulation` option added for `rtest.py`
+
+### Changed
+* `--test|-t` no longer a required flag for `rtest.py`, instead user can use either `--emulation|-e` or `--test|-t` but not both
```
Comment (amd-jnovotny) [inline CHANGELOG.md]: I added some edits to this section in #573. I think you'll want to merge those in here.

**Review by amd-jnovotny — COMMENTED** [review]
Some minor rewording for the changelog section.

File: CHANGELOG.md (hunk 2 above)
Comment (NguyenNhuDi) [reply CHANGELOG.md:10]: Hi, they are the ones that caused hardware problems in the past.

Comment (amd-jnovotny) [reply CHANGELOG.md:10]: OK, we can leave it then.

**Review by umfranzw — APPROVED** [review]
Looks good to me.

**Pull request merged**